src/compiler/checker.ts

```diff
@@ -5331,7 +5331,10 @@ namespace ts {
                         specifierCompilerOptions,
                         contextFile,
                         moduleResolverHost,
-                        { importModuleSpecifierPreference: isBundle ? "non-relative" : "relative" },
+                        {
+                            importModuleSpecifierPreference: isBundle ? "non-relative" : "relative",
+                            importModuleSpecifierEnding: isBundle ? "index" : undefined
+                        },

```

Comment:
It's probably less efficient (it constructs two different objects with different shapes), but I kind of think this is more readable:

```suggestion
                        isBundle 
                            ? { importModuleSpecifierPreference: "non-relative" }
                            : { importModuleSpecifierPreference: "relative", importModuleSpecifierEnding: "index" },
```